### PR TITLE
Implement pairing traits from crate

### PR DESCRIPTION
**Pull request by z2trillion**
To use axiom's community edition snark-verifier, we need to implement these traits again because axiom has migrated the pairing traits into an independent crate.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 mod arithmetic;
 pub mod hash_to_curve;
+// TODO: remove this and use traits defined in axiom_pairing instead.
```
Comment (roynalnaruto) [inline src/lib.rs:3]: Would this be done in a separate PR? If yes, can we file an issue for this task to keep a track?

**Review by roynalnaruto — APPROVED** [review]
Can we file an issue for the `TODO` to track its progress?

File: src/lib.rs (hunk 1 above)
Comment (z2trillion) [reply src/lib.rs:3]: I don't think we have issues enabled on this repo and I cannot see a way for me to enable them. I assume I don't have the right permissions.
